system-parachains/bridge-hubs/bridge-hub-kusama/src/bridge_to_polkadot_config.rs

```diff
@@ -246,19 +244,142 @@ impl pallet_xcm_bridge_hub::Config<XcmOverBridgeHubPolkadotInstance> for Runtime
+		// This is used to check the inbound queue/messages to determine if they can be dispatched and sent to the sibling parachain. Therefore, checking `OutXcmp` is sufficient.
+		use bp_xcm_bridge_hub_router::XcmChannelStatusProvider;
+		cumulus_pallet_xcmp_queue::bridging::OutXcmpChannelStatusProvider::<Runtime>::is_congested(with)
+	}
+
+	fn suspend_bridge(_local_origin: &Location, _: pallet_xcm_bridge_hub::BridgeId) -> Result<(), Self::Error> {
+		// IMPORTANT NOTE:
+		//
+		// Unfortunately, `https://github.com/paritytech/polkadot-sdk/pull/6231` reworked congestion is not yet released.
+		//
+		// And unfortunately, we don't have access to `XcmpQueue::send_signal(para, ChannelSignal::Suspend)` here (which would require patch release),
+		// we can add this hacky workaround/tmp/implementation that should trigger `ChannelSignal::Suspend`, e.g.:
+		/*
+		use crate::{MessageQueue, XcmpQueue};
+		use bridge_hub_common::message_queue::AggregateMessageOrigin;
+		use codec::{Decode, Encode, MaxEncodedLen};
+		use frame_support::traits::EnqueueMessage;
+		use frame_support::pallet_prelude::OptionQuery;
+		use pallet_message_queue::OnQueueChanged;
+		use scale_info::TypeInfo;
+
+		// get sibling para id
+		let local_origin_para_id: crate::ParaId = match local_origin.unpack() {
+			(1, [Parachain(id)]) => (*id).into(),
+			_ => return Err(())
+		};
+
+		// read `suspend_threshold` from `XcmpQueue` storage
+		#[derive(Copy, Clone, Eq, PartialEq, Encode, Decode, TypeInfo, MaxEncodedLen)]
+		struct QueueConfigData {
+			suspend_threshold: u32,
+			drop_threshold: u32,
+			resume_threshold: u32,
+		}
+		#[frame_support::storage_alias]
+		type QueueConfig = StorageValue<XcmpQueue, QueueConfigData, OptionQuery>;
+		let suspend_threshold = match QueueConfig::get() {
+			Some(qc) => qc.suspend_threshold,
+			None => return Err(())
+		};
+
+		// Now, this should trigger `XcmpQueue::send_signal(para, ChannelSignal::Suspend)`
+		let mut qf = MessageQueue::footprint(AggregateMessageOrigin::Sibling(local_origin_para_id));
+		qf.ready_pages = suspend_threshold;
+		XcmpQueue::on_queue_changed(local_origin_para_id.into(), qf);
+		 */
+
+		// IMPORTANT NOTE2:
+		//
+		// In the current setup, this code is likely triggered only for the hard-coded AHK<>AHP lane, as we do not support any other bridge lanes on BridgeHubs.
+		// It is triggered only when `pallet_bridge_messages::OutboundMessages` reaches 8,192 undelivered messages.
+		// The potential risk of keeping `Ok(())` or `Err(())` here is that `pallet_bridge_messages::OutboundMessages` may continue to grow:
+		//
+		// ```
+		// 	#[pallet::storage]
+		// 	pub type OutboundMessages<T: Config<I>, I: 'static = ()> =
+		// 		StorageMap<_, Blake2_128Concat, MessageKey<T::LaneId>, StoredMessagePayload<T, I>>;
+		// ```
+
+		// TODO: decide:
+		// 1. wait for patch-release stable2409-3 2024-12-12
+		// 2. go with `Ok(())` / `Err(())`
+		// 3. go with `XcmpQueue::send_signal` temporary workaround till patch release
```

Comment:
I recommend option 1: wait for and use patch-release stable2409-3 2024-12-12

If there are other time-sensitive deliverables to be deployed with this upgrade and we can't wait for `stable2409-3`, then we should do `3. go with XcmpQueue::send_signal workaround`.

I would not go with option 2 `Ok(())` / `Err(())`